arch/src/x86_64/mod.rs

```diff
@@ -92,35 +103,35 @@ pub fn configure_system(
     // Note that this puts the mptable at the last 1k of Linux's 640k base RAM
     mptable::setup_mptable(guest_mem, num_cpus).map_err(Error::MpTableSetup)?;
 
-    let mut params: boot_params = Default::default();
+    let mut params: BootParamsWrap = BootParamsWrap(boot_params::default());
 
-    params.hdr.type_of_loader = KERNEL_LOADER_OTHER;
-    params.hdr.boot_flag = KERNEL_BOOT_FLAG_MAGIC;
-    params.hdr.header = KERNEL_HDR_MAGIC;
-    params.hdr.cmd_line_ptr = cmdline_addr.offset() as u32;
-    params.hdr.cmdline_size = cmdline_size as u32;
-    params.hdr.kernel_alignment = KERNEL_MIN_ALIGNMENT_BYTES;
+    params.0.hdr.type_of_loader = KERNEL_LOADER_OTHER;
```

Comment:
Instead of always accessing the first element of the structure, you can implement deref and mutderef for BootParamsWrapper as follows:
```rust

impl std::ops::Deref for BootParamsWrapper {
    type Target = boot_params;
    fn deref(&self) -> &Self::Target {
        &self.0
    }
}

impl std::ops::DerefMut for BootParamsWrapper {
    fn deref_mut(&mut self) -> &mut boot_params {
        &mut self.0
    }
}
```

We can argue that we have to write more code, but I think that it looks better without the extra `.0` everywhere.

Reply:
Yes! This is what I was trying to do (but I was doing _something_ wrong and failed).

It does involve more code and even if it looks better it adds another layer of indirection. So I'm not on team `Deref` anymore, either way has its PROs/CONs. I'll leave you guys to decide.

Reply:
Yap, I agree that there are pros and cons for both, but what do you mean by another layer of indirection? Shouldn't it have the same effect as referencing it as a field of the wrapper struct?

Reply:
I mean that without a smart IDE that will do the `deref` for you and follow the redirection, looking at the code or trying to follow the code will be harder.

It looks better without the `.0` but I don't think it's necessarily worth it.

Reply:
If I were to choose the `Deref` method I would have to write 8 lines for each type in order to avoid using a little `.0` accessor which is actually the certified  way to access a field of a structure which is what the wrapper is. Moreover, by using the simplest and most obvious way makes the code easier to understand.
Additionally, if I implement those traits i would also have to write unit tests for them and I think we would end up investing a lot of time for an cosmetic inconvenience.